### PR TITLE
add the add button

### PR DESCRIPTION
**Pull request by NickEmma**
- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for editing task descriptions.
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique
- All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "recommendations": ["github.copilot", "GitHub.copilot"]
+}
```
Comment (arslanbisharat) [inline .vscode/extensions.json:3]: - [ ] Please remove [.vscode](https://github.com/NickEmma/todo-list/tree/dynamic/.vscode) from your project to make your project clean.

File: src/crud.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+const taskInput = document.querySelector('.input-container');
```
Comment (arslanbisharat) [inline src/crud.js:1]: - [ ] Please make sure a function for deleting a task (remove an element from the array) is implemented.

File: src/crud.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+showTodo();
+
+taskInput.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const userInput = taskHandler.value.trim();
+  taskHandler.value = '';
+  if (!todos) {
+    todos = [];
+  }
+  const task = { title: userInput, status: 'completed' };
```
Comment (arslanbisharat) [inline src/crud.js:34]: - [ ] Please make sure by default, new tasks have the property `completed` set to `false` and the property `index` set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5).

File: src/crud.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  }
+  UI.innerHTML = li;
+};
+showTodo();
+
+taskInput.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const userInput = taskHandler.value.trim();
+  taskHandler.value = '';
+  if (!todos) {
```
Comment (arslanbisharat) [inline src/crud.js:31]: - [ ] Please make sure deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @NickEmma, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 5]
```diff
@@ -2,45 +2,47 @@
...
+//     description: 'read my book',
+//     completed: false,
+//     index: 3,
+//   },
+//   {
+//     description: 'working on my coding skills',
+//     completed: false,
+//     index: 4,
+//   },
+// ];
```
[the remark this answers is not in the document]
Comment (NickEmma) [reply src/index.js:27]: okay

File: .vscode/extensions.json (hunk 1 above)
Comment (NickEmma) [reply .vscode/extensions.json:3]: noted

File: src/crud.js (hunk 3 above)
Comment (NickEmma) [reply src/crud.js:34]: resolved

File: src/index.js [hunk 6]
```diff
@@ -2,45 +2,47 @@
...
+//     <div class="icon">
+//     <i class="fa-solid fa-ellipsis-vertical"></i>
+//     </div>
+//     `;
+//     content.appendChild(todo);
+//   }
+// };
 
-data();
+// data();
```
[the remark this answers is not in the document]
Comment (NickEmma) [reply src/index.js:48]: resolved

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @NickEmma 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**